install.ps1

```diff
@@ -0,0 +1,132 @@
+        return $InstallDir
+    }
+
+    throw "Invalid install directory provided '$InstallDir'"
+}
+
+function Get-Temp-Directory() {
+    return $env:temp
+}
+
+function Prepare-Install-Directory([string]$InstallDir) {
+    New-Item -ItemType Directory -Force -Path $InstallDir | Out-Null
+}
+
+function Download-OpenTelemetry([string]$Version, [string]$Path) {
+    $archive = "opentelemetry-dotnet-instrumentation-windows.zip"
+    $dlUrl = "https://github.com/open-telemetry/opentelemetry-dotnet-instrumentation/releases/download/$Version/$archive"
+    $dlPath = Join-Path $Path $archive
+
+    Invoke-WebRequest -Uri $dlUrl -OutFile $dlPath
+
+    return $dlPath
+}
+
+function Extract-OpenTelemetry([string]$DlPath, [string] $InstallPath) {
+    Expand-Archive $DlPath $InstallPath -Force
+}
+
+function Setup-OpenTelemetry-Environment([string] $InstallPath) {
+    $target = [System.EnvironmentVariableTarget]::Machine
+
+    # .NET Framework
+    [System.Environment]::SetEnvironmentVariable('COR_PROFILER','{918728DD-259F-4A6A-AC2B-B85E1B658318}', $target)
+    [System.Environment]::SetEnvironmentVariable('COR_PROFILER_PATH_32', (Join-Path $InstallPath "/win-x86/OpenTelemetry.AutoInstrumentation.Native.dll"), $target)
+    [System.Environment]::SetEnvironmentVariable('COR_PROFILER_PATH_64', (Join-Path $InstallPath "/win-x64/OpenTelemetry.AutoInstrumentation.Native.dll"), $target)
+
+    # .NET Core
+    [System.Environment]::SetEnvironmentVariable('CORECLR_PROFILER','{918728DD-259F-4A6A-AC2B-B85E1B658318}', $target)
+    [System.Environment]::SetEnvironmentVariable('CORECLR_PROFILER_PATH_32', (Join-Path $InstallPath "/win-x86/OpenTelemetry.AutoInstrumentation.Native.dll"), $target)
+    [System.Environment]::SetEnvironmentVariable('CORECLR_PROFILER_PATH_64', (Join-Path $InstallPath "/win-x64/OpenTelemetry.AutoInstrumentation.Native.dll"), $target)
+
+    # .NET Common
+    [System.Environment]::SetEnvironmentVariable('DOTNET_ADDITIONAL_DEPS', (Join-Path $InstallPath "AdditionalDeps"), $target)
+    [System.Environment]::SetEnvironmentVariable('DOTNET_SHARED_STORE', (Join-Path $InstallPath "store"), $target)
+    [System.Environment]::SetEnvironmentVariable('DOTNET_STARTUP_HOOKS', (Join-Path $InstallPath "netcoreapp3.1/OpenTelemetry.AutoInstrumentation.StartupHook.dll"), $target)
+
+    # ASP.NET Core
+    [System.Environment]::SetEnvironmentVariable('ASPNETCORE_HOSTINGSTARTUPASSEMBLIES', "OpenTelemetry.AutoInstrumentation.AspNetCoreBootstrapper", $target)
+
+    # OpenTelemetry
+    [System.Environment]::SetEnvironmentVariable('OTEL_DOTNET_AUTO_HOME', $InstallPath, $target)
+    [System.Environment]::SetEnvironmentVariable('OTEL_DOTNET_AUTO_INTEGRATIONS_FILE', (Join-Path $InstallPath "integrations.json"), $target)
+}
+
+function Setup-IIS-Environment() {
+    Setup-Windows-Service -ServiceName "W3SVC"
+    Setup-Windows-Service -ServiceName "WAS"
+}
+
+function Setup-Windows-Service([string]$ServiceName) {
```

Comment:
It does not set `OTEL_SERVICE_NAME` for the instrumented service.

I think it will be easier to provide separate `instrument-service.ps1` + `uninstriument-service.ps1` scripts. 

I suggest tackling Windows Services + IIS instrumentation in a separate PR.

Reply:
This just setups IIS and could be the base for windows service setup, if there is good idea for UX. (likely object array param that setups each one)